src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksUploadSuppliersServiceImpl.java

```diff
@@ -0,0 +1,220 @@
+    }
+
+    private void performSupplierUploadProcessing(Collection<PaymentWorksVendor> vendors) {
+        PaymentWorksUploadSuppliersBatchReportData reportData = new PaymentWorksUploadSuppliersBatchReportData();
+        reportData.getRecordsFoundToProcessSummary().setRecordCount(vendors.size());
+        boolean uploadSucceeded = uploadVendorsToPaymentWorks(vendors, reportData);
+        updateVendorStatusesInKfs(vendors, uploadSucceeded, reportData);
+        paymentWorksUploadSuppliersReportService.generateAndEmailProcessingReport(reportData);
+    }
+
+    protected boolean uploadVendorsToPaymentWorks(Collection<PaymentWorksVendor> vendors, PaymentWorksUploadSuppliersBatchReportData reportData) {
+        InputStream vendorCsvDataStream = null;
+        
+        try {
+            checkForPotentialVendorDeletionAttempts(vendors);
+            byte[] vendorCsvData = generateVendorCsvDataForUpload(vendors);
+            vendorCsvDataStream = new ByteArrayInputStream(vendorCsvData);
+            int receivedVendorCount = paymentWorksWebServiceCallsService.uploadVendorsToPaymentWorks(vendorCsvDataStream);
+            reportData.getRecordsProcessedByPaymentWorksSummary().setRecordCount(receivedVendorCount);
+            if (receivedVendorCount != vendors.size()) {
+                LOG.warn("uploadVendorsToPaymentWorks, " + vendors.size()
+                        + " vendors were sent to PaymentWorks, but PaymentWorks reported that it received "
+                        + receivedVendorCount + " vendors instead");
+                reportData.getGlobalMessages().add(paymentWorksBatchUtilityService.retrievePaymentWorksParameterValue(
+                        PaymentWorksParameterConstants.PAYMENTWORKS_UPLOAD_SUPPLIERS_REPORT_VENDOR_COUNT_MISMATCH_MESSAGE));
+            }
+            return true;
+        } catch (RuntimeException e) {
+            LOG.error("uploadVendorsToPaymentWorks, error encountered when uploading vendors to PaymentWorks", e);
+            reportData.getGlobalMessages().add(paymentWorksBatchUtilityService.retrievePaymentWorksParameterValue(
+                    PaymentWorksParameterConstants.PAYMENTWORKS_UPLOAD_SUPPLIERS_REPORT_VENDOR_UPLOAD_FAILURE_MESSAGE));
+            return false;
+        } finally {
+            IOUtils.closeQuietly(vendorCsvDataStream);
+        }
+    }
+
+    private void checkForPotentialVendorDeletionAttempts(Collection<PaymentWorksVendor> vendors) {
+        boolean vendorDeletionAttemptExists = vendors.stream()
+                .anyMatch((vendor) -> StringUtils.equalsIgnoreCase(
+                        PaymentWorksConstants.SUPPLIER_UPLOAD_DELETE_INDICATOR, vendor.getRequestingCompanyLegalNameForProcessing()));
+        if (vendorDeletionAttemptExists) {
+            LOG.error("checkForPotentialVendorDeletionAttempts, discovered a potential attempt to delete a vendor via the supplier upload");
```

Comment:
If needed, I could restructure this search so that it returns the actual problematic vendors so that they can be logged. However, all of the attempted vendors should be getting included in the generated report anyway, so it should be simple to find the ones that were causing problems.